### ISSUE TITLE
Spike: Encompassing error handling

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#17]
### Expected result

What would be a catch-all error handling strategy? Apart from the UseExceptionHandler middleware, should we use UseStatusCodePagesWithReExecute? Given the different kinds of errors being trapped, what would be the ideal mix and order?

How can we log all possible errors into the Xperience event log? How to deal with app startup errors, when the DB is still not available?

### Additional context

Add any other context or guidelines here.

### Resources

* https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-3.1
* https://stackoverflow.com/questions/46373668/when-using-usestatuscodepageswithreexecute-statuscode-is-not-sent-to-browser

### Issue expands upon

This issue expands upon another issue(s): #10 

### Blockers

This issue is blocked by the following issue(s): #`issue number`
